packages/components/src/text-highlight/index.js

```diff
@@ -16,13 +16,17 @@ const TextHighlight = ( { text = '', highlight = '' } ) => {
 	}
 
 	const regex = new RegExp( `(${ escapeRegExp( highlight ) })`, 'gi' );
-	const parts = text.split( regex );
-	return (
-		<Fragment>
-			{ parts.filter( ( part ) => part ).map( ( part, i ) => (
-				regex.test( part ) ? <mark key={ i }>{ part }</mark> : <span key={ i }>{ part }</span>
-			) ) }
-		</Fragment>
+	const parts = text.split( regex ).filter( ( part ) => part );
```

Comment:
Could it use [String.replace](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace) with the RegExp or do I miss something specific in the logic?

Reply:
Actually I think it's fine. Probably me overengineering things.